src/main/java/core/basesyntax/figures/Figure.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax.figures;
+
+import core.basesyntax.AreaCalculator;
+import core.basesyntax.InfoPrinter;
+
+public abstract class Figure implements AreaCalculator, InfoPrinter {
+    private String name;
```

Comment:
Having this property is not a good idea. In your code when you create Circle you always need to set name to "circle" etc. 
1. It's a risk to get Circle with name "Square" for example.
2. It's data duplication and you waste memory storing all these names for each figure.

It's possible to use getClass().getSimpleName() to have propper string value.